sharding/data_modeling.rst

```diff
@@ -0,0 +1,234 @@
+
+Let's examine typical real-time schemas.
+
+Raw Events Table
+----------------
+
+In this scenario we ingest high volume sensor measurement events into a single table and distribute it across Citus by the :code:`device_id` of the sensor. Every time the sensor makes a measurement we save that as a single event row with measurement details in a jsonb column for flexibility.
+
+.. code-block:: postgres
+
+  CREATE TABLE events (
+    device_id bigint NOT NULL,
+    event_id uuid NOT NULL,
+    event_time timestamptz NOT NULL,
+    event_type int NOT NULL,
+    payload jsonb,
+    PRIMARY KEY (device_id, event_id)
+  );
+  CREATE INDEX ON events USING BRIN (event_time);
+
+  SELECT create_distributed_table('events', 'device_id');
+
+Any query that restricts to a given device is routed directly to a worker node for processing. We call this a *single-shard* query. Here is one to get the ten most recent events:
+
+.. code-block:: postgres
+
+  SELECT event_time, payload
+    FROM events
+    WHERE device_id = 298
+    ORDER BY event_time DESC
+    LIMIT 10;
+
+To take advantage of massive parallelism we can run a *cross-shard* query. For instance, we can find the min, max, and average temperatures per minute across all sensors in the last ten minutes (assuming the json payload includes a :code:`temp` value). We can scale this query to any number of devices by adding worker nodes to the Citus cluster.
+
+.. code-block:: postgres
+
+  SELECT minute,
+    min(temperature)::decimal(10,1) AS min_temperature,
+    avg(temperature)::decimal(10,1) AS avg_temperature,
+    max(temperature)::decimal(10,1) AS max_temperature
+  FROM (
+    SELECT date_trunc('minute', event_time) AS minute,
+           (payload->>'temp')::float AS temperature
+    FROM events
+    WHERE event_t1me >= now() - interval '10 minutes'
+  ) ev
+  GROUP BY minute
+  ORDER BY minute ASC;
+
+Events and Summaries
+--------------------
+
+The previous example calculates statistics at runtime, doing possible recalculation between queries. Another approach is precalculating aggregates. This avoids recalculating raw event data and results in even faster queries. For example, a web analytics dashboard might want a count of views per page per day. The raw events data table looks like this:
+
+.. code-block:: postgres
+
+  CREATE TABLE page_views (
+    page_id int PRIMARY KEY,
+    host_ip inet,
+    view_time timestamp default now()
+  );
+  CREATE INDEX view_time_idx ON page_views USING BRIN (view_time);
+
+  SELECT create_distributed_table('page_views', 'page_id');
+
+We will precompute the daily view count in this summary table:
+
+.. code-block:: postgres
+
+  CREATE TABLE daily_page_views (
+    day date,
+    page_id int,
+    view_count bigint,
+    PRIMARY KEY (day, page_id)
+  );
+
+  SELECT create_distributed_table('daily_page_views', 'page_id',
+                                  colocate_with => 'page_views');
```

Comment:
For the examples in this page, we don't need to show the ``colocate_with`` option.

For simplicity, could we remove ``colocate_with`` from this section and the next?